plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php

```diff
@@ -188,11 +188,14 @@ public function add_settings_page( $pages ) {
 		 * @return array
 		 */
 		public function add_settings_page_data( $pages ) {
+			global $current_section;
+
 			$sections      = $this->get_sections();
 			$sections_data = array();
 
 			// Loop through each section and get the settings for that section.
 			foreach ( $sections as $section_id => $section_label ) {
+				$current_section       = $section_id;
```

Comment:
The $current_section value will remain set to the last section's ID after the method completes. Do we need to reset it at the end of the method?

Reply:
Thats probably safest. Done in [Reset the current section to the saved current section.](https://github.com/woocommerce/woocommerce/pull/55712/commits/07321dab3db9e921bc65a7f37b04774b24511c9c)